resources/api_define_trigger_set.ts

```diff
@@ -0,0 +1,23 @@
+import { RaidbossData } from '../types/data';
+import { TriggerSet } from '../types/trigger';
+
+export function defineTriggerSet<InitData>(
+  set:
+    & TriggerSet<RaidbossData & InitData>
+    & (
+      Record<string, never> extends InitData ? Record<never, never>
+        : { initData: () => InitData }
+    ),
+): TriggerSet<RaidbossData>;
+
+export function defineTriggerSet(
+  set: Omit<TriggerSet<RaidbossData>, 'initData'>,
+): TriggerSet<RaidbossData>;
+
+/* eslint-disable func-style */
+export function defineTriggerSet<Data extends RaidbossData>(
+  set: TriggerSet<Data>,
+): TriggerSet<Data> {
+  return set;
+}
+/* eslint-enable func-style */
```

Comment:
Is there a reason that something like this wouldn't work instead, and not require disabling the eslint rule?
```ts
export defineTriggerSet<Data extends RaidbossData>(
  set: TriggerSet<Data>,
): TriggerSet<Data> => {
  return set;
}
```

Reply:
When to declare a function with overrides, I think this approach would be much clear. Every overrides would be regarded as a function declaration.
```ts
export function defineTriggerSet(set): TriggerSet<Data>;
export function defineTriggerSet<...>(set: TriggerSet<...>): TriggerSet<Data>;
export function defineTriggerSet<Data>(set: TriggerSet<Data>): TriggerSet<Data> {
  // ...
}
```

Rather than regarded as a type?
```ts
export const defineTriggerSet: {
  <InitData>(
    set:
      & TriggerSet<RaidbossData & InitData>
      & (
        Record<string, never> extends InitData ? Record<never, never>
          : { initData: () => InitData }
      ),
  ): TriggerSet<RaidbossData>;

  (set: Omit<TriggerSet<RaidbossData>, 'initData'>): TriggerSet<RaidbossData>;
} = <Data extends RaidbossData>(
  set: TriggerSet<Data>,
): TriggerSet<Data> => {
  return set;
};
```